File: src/jsjava/security/AccessController.java
package jsjava.security;

import java.io.InputStream;

public class AccessController implements AccessControlContext {

	// a dummy class
	
	public static <T> T doPrivileged(PrivilegedAction<T> action) {
		return action.run();
	}

	public static AccessControlContext getContext() {
		return new AccessController();
	}

	public boolean checkPermission(Object perm) {
		// no access checking in JavaScript
		return true;
	}

  public static <T> T doPrivileged(PrivilegedAction<T> action,
      AccessControlContext context) {
  	return action.run();
  }

}

File: site/swingjs/j2s/jsjava/security/AccessController.js
Clazz.declarePackage ("jsjava.security");
Clazz.load (["jsjava.security.AccessControlContext"], "jsjava.security.AccessController", null, function () {
c$ = Clazz.declareType (jsjava.security, "AccessController", null, jsjava.security.AccessControlContext);
c$.doPrivileged = Clazz.defineMethod (c$, "doPrivileged", 
function (action) {
return action.run ();
}, "jsjava.security.PrivilegedAction");
c$.getContext = Clazz.defineMethod (c$, "getContext", 
function () {
return  new jsjava.security.AccessController ();
});
Clazz.overrideMethod (c$, "checkPermission", 
function (perm) {
return true;
}, "~O");
c$.doPrivileged = Clazz.defineMethod (c$, "doPrivileged", 
function (action, context) {
return action.run ();
}, "jsjava.security.PrivilegedAction,jsjava.security.AccessControlContext");
});
